tools/passive_extraction_workflow_latest.py

```
@@ -5109,18 +5119,19 @@ async def _extract_supplier_products(
-                entered_url = normalize_url(category_url)
                 sp["current_category_url"] = entered_url
-
-                pci = int(sp.get("persistent_category_index", 0))
-                expected_url = normalize_url(category_urls[pci]) if 0 <= pci < len(category_urls) else None
+                # Hard-align PCI to the absolute index for determinism across runs/batches
+                sp["persistent_category_index"] = int(absolute_cat_index)
+                self.state_manager.save_state_atomic("align-pci-to-absolute")
+                # Guard: URL of PCI must match the entered URL
+                try:
+                    expected_url = normalize_url(category_urls[sp["persistent_category_index"]])
+                except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)